ScoutSuite/providers/aws/resources/iam/service.py

```diff
@@ -0,0 +1,134 @@
+import asyncio
+
+from ScoutSuite.providers.aws.resources.resources import AWSCompositeResources
+from ScoutSuite.providers.aws.resources.iam.credentialreports import CredentialReports
+from ScoutSuite.providers.aws.resources.iam.groups import Groups
+from ScoutSuite.providers.aws.resources.iam.policies import Policies
+from ScoutSuite.providers.aws.resources.iam.users import Users
+from ScoutSuite.providers.aws.resources.iam.roles import Roles
+from ScoutSuite.providers.aws.resources.iam.passwordpolicy import PasswordPolicy
+from ScoutSuite.providers.aws.facade.facade import AWSFacade
+
+
+class IAM(AWSCompositeResources):
+    _children = [
+        (CredentialReports, 'credential_reports'),
+        (Groups, 'groups'),
+        (Policies, 'policies'),
+        (Users, 'users'),
+        (Roles, 'roles'),
+        (PasswordPolicy, 'password_policy')
+    ]
+
+    def __init__(self):
+        # TODO: Should be injected
+        self.facade = AWSFacade()
+        self.service = 'iam'
+
+    async def fetch_all(self, credentials, regions=None, partition_name='aws'):
+        # TODO: This should not be set here, the facade should be injected and already authenticated
+        self.facade._set_session(credentials)
+        await self._fetch_children(self, {})
+
+        # We do not want the report to count the password policies as resources, they aren't really resources.
+        self['password_policy_count'] = 0
+
+    async def finalize(self):
```

Comment:
I think this function could be non-async?

Reply:
It could indeed, but it is awaited (in the base provider I think), so if I remove the async, we'll get a _"can't await None"_ or some error like that.